pkgs/by-name/sy/syncat/package.nix

```diff
@@ -0,0 +1,28 @@
+{ lib, rustPlatform, fetchFromGitHub, openssl, libgit2, pkg-config }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "syncat";
+  version = "3.4.0";
+
+  src = fetchFromGitHub {
+    owner = "foxfriends";
+    repo = "syncat";
+    rev = version;
+    hash = "sha256-tjl8qdC2MAGA5ap+ZOPoY7fubsFk7YMy3JVR77s6mxk=";
+  };
+
+  cargoHash = "sha256-9DqdAcCfebCQwgF5R5ZXaO5Y7uNnNyHznQG7BgIKgYo=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl libgit2 ];
+
+  doCheck = false;
```

Comment:
Would you mind adding a short comment, describing why tests were disabled?

Reply:
I disabled it because of this output from the checkPhase (:

### Output from tests enabled:
```
Executing cargoCheckHook
++ cargo test -j 12 --release --target x86_64-unknown-linux-gnu --frozen -- --test-threads=12
   Compiling tree-sitter v0.20.10
   Compiling openssl-sys v0.9.93
   Compiling libz-sys v1.1.12
   Compiling syncat-stylesheet v2.4.0 (/build/source/syncat-stylesheet)
   Compiling libloading v0.5.2
   Compiling syncat-stylesheet v2.4.0
   Compiling libssh2-sys v0.2.23
   Compiling libgit2-sys v0.14.2+1.5.1
   Compiling git2 v0.16.1
   Compiling syncat v3.4.0 (/build/source/syncat)
    Finished release [optimized] target(s) in 17.49s
     Running unittests src/main.rs (target/x86_64-unknown-linux-gnu/release/deps/syncat-fb39913fcb947a33)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests src/lib.rs (target/x86_64-unknown-linux-gnu/release/deps/syncat_stylesheet-4d1c0d2d45b18adf)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

   Doc-tests syncat-stylesheet

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

Finished cargoCheckHook
```

Reply:
But I can just remove it if you think it is incorrect to disable it?

Reply:
by default we should enable tests unless we can't make the package build otherwise. (and that is in the manual)

Reply:
Right! If the author of the package ever chooses to make any tests, we'll probably want to reenable them. Maybe you could add a comment saying you disabled them because there were no tests at the time of creating the package?

Reply:
Good idea. Give me a sec and I add a comment.

Reply:
@eclairevoyant - again I found documentation that says otherwise.. [https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md#disabling-package-tests-disabling-package-tests](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md#disabling-package-tests-disabling-package-tests)

What is the correct solution?

Reply:
hmm if they explicitly say that then I can't argue with it. A comment would be good then.

Reply:
The original reasoning I had in mind was, if upstream adds tests, we won't know about it if it's disabled, and I doubt someone checks manually on every version bump. Seems like an additional maintenance burden imo.

Reply:
Cool, thanks. Still trying to get the hang of that is valid documentation and what not. I appreciate the help 😃